presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -419,9 +420,9 @@ public int nextBatch()
         // single fixed width column are: 1, 16, 256, 1024, 1024,..., 1024, 256 and the 256 was because there is only
         // 256 rows left in this row group, then the nextBatchSize should be 1024 instead of 512. So we need to grow the
         // nextBatchSize before limiting the currentBatchSize by currentGroupRowCount - nextRowInGroup.
-        currentBatchSize = toIntExact(min(nextBatchSize, maxBatchSize));
-        nextBatchSize = min(currentBatchSize * BATCH_SIZE_GROWTH_FACTOR, MAX_BATCH_SIZE);
-        currentBatchSize = toIntExact(min(currentBatchSize, currentGroupRowCount - nextRowInGroup));
+        currentBatchSize = min(nextBatchSize, maxBatchSize);
+        nextBatchSize = min(multiplyExact(currentBatchSize, BATCH_SIZE_GROWTH_FACTOR), MAX_BATCH_SIZE);
```

Comment:
@elonazoulay These changes make sense, but it is not clear whether they fix anything. I expect `multiplyExact` to throw a clear error message on overflow, but not necessary avoid the overflow. What happened before this change and what is happening now?

Reply:
The overflow would be ignored and multiplying currentBatchSize by BATCH_SIZE_GROWTH_FACTOR could result in silent overflow (i.e. the number would be negative and no exception thrown).

Reply:
@elonazoulay Makes sense. Thanks for explaining.